spec/lib/twingly/url_spec.rb

```diff
@@ -70,12 +70,30 @@ def valid_urls
 end
 
 def leading_and_trailing_whitespace
+  line_feed          = "\u000A"
+  carriage_return    = "\u000D"
+  non_breaking_space = "\u00A0"
+  space              = "\u0020"
+
   {
-    "non-breaking space and space" => "\u00A0\u0020",
-    "non-breaking space" => "\u00A0",
-    "non-breaking space, space, non-breaking space" => "\u00A0\u0020\u00A0",
-    "space and non-breaking space" => "\u0020\u00A0",
-    "space, non-breaking space and space" => "\u0020\u00A0\u0020",
+    "non-breaking space and space"                  => [non_breaking_space, space].join,
```

Comment:
I find the following easier to understand:

```suggestion
    "non-breaking space and space"                  => "#{non_breaking_space}#{space}"
```

(and for the rest)

Reply:
Oh no 🙈 I started like that but thought it was too fiddly and was very happy when I came up with the join approach :-)

Reply:
Hehe, yes it's prettier, but I found myself pondering "what's the default behavior on join"

Reply:
I know it is still basically the same, but what if we DRY the `#join`, does it help or does it makes it even worse? :-)

```ruby
...  
    "carriage-return and line-feed"                 => [carriage_return, line_feed],
    "line-feed and carriage-return"                 => [line_feed, carriage_return],
  }.map { |key, val| [key, val.join] }.to_h

Reply:
I thought it was easier to both read and write the arrays... seeing we will/should generate the combinations if we add more, the "easier to write" might not matter much.

Reply:
> Hehe, yes it's prettier, but I found myself pondering "what's the default behavior of #join"

Hehe, yeah, I've taught myself what it does :-) It is very hand I've come to realize.

Reply:
We can leave it like this. Let's skip the DRY part IMO :)